R/peaks-functions.R

```diff
@@ -429,3 +429,65 @@ joinPeaksGnps <- function(x, y, xPrecursorMz = NA_real_,
     x[, 2L] <- MsCoreUtils::smooth(x[, 2L], cf = coef)
     x
 }
+
+#' @description
+#'
+#' Removes (Orbitrap) FFT artefacts (peaks) from a spectrum keeping peaks
+#' potentially representing [13]C isotopes.
+#'
+#' @param x peaks matrix
+#'
+#' @param halfWindowSize `numeric(1)` defining the (m/z) window left and right
+#'     of a peak to look for artefacts.
+#'
+#' @param threshold `numeric(1)` defining the relative intensity to the actual
+#'     peak below which peaks are considered artefacts. Defaults to
+#'     `threshold = 0.2` hence removing all peaks with an intensity below 0.2 *
+#'     the peak intensity.
+#'
+#' @param keepIsotopes `logical(1)` whether it should be screened for potential
+#'     isotope peaks. Defaults to `keepIsotopes = TRUE` thus candidate artefact
+#'     peaks potentially being isotopes will **not** be removed.
+#'
+#' @param maxCharge `integer(1)`
+#'
+#' @param isotopeTolerance `numeric(1)`
+#'
+#' @author Jan Stanstrup
+#'
+#' @importFrom MsCoreUtils between
```

Comment:
`MsCoreUtils::common` is needed for the suggested `find_isotopes` part. `MsCoreUtils::between` could removed if the suggested alternative implementation is accepted.

```suggestion
#' @importFrom MsCoreUtils common
```